pilot/pkg/model/context.go

```diff
@@ -600,6 +600,10 @@ const (
 
 	// NodeMetadataTLSClientRootCert is the absolute path to client root cert file
 	NodeMetadataTLSClientRootCert = "TLS_CLIENT_ROOT_CERT"
+
+	// NodeMetadataIdleTimeout specifies the idle timeout for the proxy, in duration format (10s).
+	// If not set, no timeout is set.
+	NodeMetadataIdleTimeout = "IDLE_TIMEOUT"
```

Comment:
Why add a new env, I remember a timeout field is added to api. You can check.

Reply:
What api? If you mean destination rule I believe that timeout with not work, ee need to close connections even before routing happens (eg if they open connection and send nothing). If you mean https://github.com/istio/api/pull/879/files then I think we should close that, we should have a proper fix in 1.2 in the Sidecar, or even better use Shrirams change that allows configuring generic envoy fields so we don't have to handle every possible envoy option. But for now in 1.1 we just need a quick solution to get this change in , so env is an easy backwards compatible way to do so

Reply:
Yeah, I mean destinationRule

Reply:
I don't think that will work, we need timeout at the top level, before we route to a destination and destination rule takes place. I did not test that the dest rule setting will not satisfy this, but I will tomorrow.

Reply:
Destination rule setting will not work here. That goes into the cluster properties.
Here we want listener (hcm) properties and we have not modeled anywhere yet.